### PR TITLE
feat: add code examples

### PR DESCRIPTION
**Pull request by giltinde**
The following appears in the terminal when running locally:
```
INFO:     127.0.0.1:50537 - "GET /9e68c24da9f5fd56991c.worker.js.map HTTP/1.1" 404 Not Found
```
But it's maybe not an issue: https://github.com/Redocly/redoc/issues/1240

### CODE REVIEW

**Comment by dymbe** [comment]
> The following appears in the terminal when running locally:
> 
> ```
> INFO:     127.0.0.1:50537 - "GET /9e68c24da9f5fd56991c.worker.js.map HTTP/1.1" 404 Not Found
> ```
> 
> But it's maybe not an issue: [Redocly/redoc#1240](https://github.com/Redocly/redoc/issues/1240)

It's normal if it only happens when you have to console open

File: flood_api/example_code/curl/detailed.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Get the detailed flood forecast for the given coordinates and neighboring cells, between 2023-12-01 and 2023-12-07 (inclusive)
+curl -i -X GET $api_url$api_path?lon=22.260536&lat=4.882569&include_neighbors=true&start_date=2023-12-01&end_date=2023-12-07
+
+# Get the detailed flood forecast for the given bounding box without neighboring cells, covering the entire forecast duration
+curl -i -X GET $api_url$api_path?min_lon=22.0&max_lon=23.05&min_lat=4.764412&max_lat=5.015732
```
Comment (kstigen) [inline flood_api/example_code/curl/detailed.sh:5]: Add newlines in all files

File: flood_api/openapi/openapi.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                example_code_dir / lang.lower() / f"{route.name}.{file_ext}"
+            )
+            if os.path.isfile(file_with_code_sample):
+                with open(file_with_code_sample) as f:
+                    code_template = Template(f.read())
+                    code_samples.append(
+                        {
+                            "lang": lang,
+                            "source": code_template.safe_substitute(
+                                api_url=f"{settings.api_url}",
```
Comment (kstigen) [inline flood_api/openapi/openapi.py:45]: We can simplify all the example-code files by concatenating `api_url` and `api_path` here.

File: flood_api/example_code/python/threshold.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from httpx import Client
+
+with Client() as client:
+    # Get the return period thresholds for the given coordinates
+    response_loc = client.get(
+        url="$api_url$api_path", params={"lon": 22.260536, "lat": 4.882569}
```
Comment (kstigen) [inline flood_api/example_code/python/threshold.py:6]: Simplify all the example-code files by only requiring a single parameter to be replaced. Rename to `endpoint_url`. See comment in `openapi.py`

File: flood_api/openapi/openapi.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+supported_languages = {"cURL": "sh", "JavaScript": "js", "Python": "py"}
+
+
+def custom_openapi(app: FastAPI, example_code_dir: Path):
+    if app.openapi_schema:
+        return app.openapi_schema
+
+    openapi_schema = get_openapi(
+        title="Geocoder API",
```
Comment (kstigen) [inline flood_api/openapi/openapi.py:20]: Copy-paste error

File: flood_api/openapi/openapi.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        version=settings.version,
+        description=settings.api_description,
+        routes=app.routes,
+    )
+
+    openapi_schema["info"]["x-logo"] = {
+        "url": "https://api-test.openepi.io/assets/icons/open-epi-logo.svg"
```
Comment (kstigen) [inline flood_api/openapi/openapi.py:27]: See weather-api for how to construct an url to given environment

File: flood_api/__main__.py [hunk 6]
```diff
@@ -22,12 +25,27 @@ async def lifespan(flood_app: FastAPI):
...
+
+example_code_dir = pathlib.Path(__file__).parent / "example_code"
+app.openapi_schema = openapi.custom_openapi(app, example_code_dir)
+
+
+@app.get("/redoc", include_in_schema=False)
+def redoc():
+    return get_redoc_html(
+        openapi_url="/openapi.json",
+        title="Geocoder API",
```
Comment (kstigen) [inline flood_api/__main__.py:44]: Copy-paste error

File: flood_api/__main__.py [hunk 7]
```diff
@@ -22,12 +25,27 @@ async def lifespan(flood_app: FastAPI):
     version=settings.version,
```
Comment (kstigen) [inline flood_api/__main__.py:25]: `version` and `description` are redundant here, as we redefine them in `openapi.py`. Can be removed

**Review by kstigen — CHANGES_REQUESTED** [review]

**Pull request merged**